http-verbs-common/src/main/scala/uk/gov/hmrc/http/HttpTransport.scala

```diff
@@ -101,13 +103,25 @@ trait HttpTransport
     with PostHttpTransport {}
 
 trait CoreGet {
+  def GET[A](
+    url: URL,
+    headers: Seq[(String, String)] = Seq.empty)(implicit rds: HttpReads[A], hc: HeaderCarrier, ec: ExecutionContext): Future[A]
+
   def GET[A](
     url: String,
     queryParams: Seq[(String, String)],
     headers: Seq[(String, String)])(
       implicit rds: HttpReads[A],
       hc: HeaderCarrier,
-      ec: ExecutionContext): Future[A]
+      ec: ExecutionContext): Future[A] = {
+    if (queryParams.nonEmpty && url.contains("?")) {
+      throw new UrlValidationException(
+        url,
+        s"${this.getClass}.GET(url, queryParams)",
+        "Query parameters should be provided in either url or as a Seq of tuples")
+    }
+    GET(url"$url?$queryParams", headers)
```

Comment:
does look quite slick! seems like it would be pretty easy for teams just to add `url`to the front of their strings and reap the benefits